packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -105,6 +121,153 @@ export default {
         componentScope = currentScope;
       }
 
+      // Next we'll define a few helpers that helps us
+      // tell if some values don't have to be declared as deps.
+
+      // Some are known to be static based on Hook calls.
+      // const [state, setState] = useState() / React.useState()
+      //               ^^^ true for this reference
+      // const [state, dispatch] = useReducer() / React.useReducer()
+      //               ^^^ true for this reference
+      // const ref = useRef()
+      //       ^^^ true for this reference
+      // False for everything else.
+      function isStaticKnownHookValue(resolved) {
```

Comment:
This function just moved into the closure. It's the same implementation with a slightly different argument.